lib/features/family/features/reflect/domain/grateful_recommendations_repository_impl.dart

```diff
@@ -1,28 +1,49 @@
 import 'package:givt_app/features/family/features/recommendation/organisations/models/organisation.dart';
-import 'package:givt_app/features/family/features/recommendation/organisations/repositories/organisations_repository.dart';
 import 'package:givt_app/features/family/features/reflect/domain/grateful_recommendations_repository.dart';
 import 'package:givt_app/features/family/features/reflect/domain/models/game_profile.dart';
+import 'package:givt_app/features/family/network/api_service.dart';
 
 class GratefulRecommendationsRepositoryImpl
     implements GratefulRecommendationsRepository {
+  GratefulRecommendationsRepositoryImpl(
+    this._familyApiService,
+  );
 
-  GratefulRecommendationsRepositoryImpl(this._organisationsRepository);
-
-  final OrganisationsRepository _organisationsRepository;
+  final FamilyAPIService _familyApiService;
   final Map<GameProfile, List<Organisation>> _gratefulRecommendations = {};
 
   @override
   Future<Map<GameProfile, List<Organisation>>>
       getGratefulRecommendationsForMultipleProfiles(
     List<GameProfile> profiles,
   ) async {
-    throw UnimplementedError();
+    final results = await Future.wait(
+      profiles.map((profile) async {
```

Comment:
I would either put this in an all-encompasing try catch or the usage of it.
Also if you do it here I would not let it return anything and rename the method like we discussed in the previous PR.

Reply:
Some error handling is added, method renamed and fetching just one as fallback is also improved :)